src/pkg/cli/compose/loader.go

```diff
@@ -120,6 +137,9 @@ func (c *Loader) newProjectOptions() (*cli.ProjectOptions, error) {
 		cli.WithDiscardEnvFile,
 		cli.WithConsistency(false), // TODO: check fails if secrets are used but top-level 'secrets:' is missing
 		cli.WithLoadOptions(func(o *loader.Options) {
+			o.Interpolate = &interpolation.Options{}
+			o.Interpolate.LookupValue = interpolationConfig.LookupEnv
+
```

Comment:
Previous sdk (2.1.4), **_o.interpolate_** was already populated as was the LookupValue which had environment variables already loaded. Because we do not have access to the environment variables we need to get it from after the project loads. We then call loadProject() again with the new environment variables loaded into the struct which has LookupEnv() for our custom substitution.